File: ruby/raindrops/raindrops.rb
module BookKeeping
  VERSION = 3
end

class Raindrops
  def self.convert(num)
    final = ''
    if num % 3 == 0
      final += 'Pling'
    end
    if num % 5 == 0
      final += 'Plang'
    end
    if num % 7 == 0
      final += 'Plong'
    end
    if final == ''
      final = num.to_s
    end
    final
  end
end

File: ocaml/raindrops/raindrops.ml
let raindrop num =
  let result = "" in
  let result = if num mod 3 == 0 then "Pling" else result in
  let result = if num mod 5 == 0 then result ^ "Plang" else result in
  let result = if num mod 7 == 0 then result ^ "Plong" else result in
  if result = "" then string_of_int num else result;;
